R/utils.R

```diff
@@ -200,27 +200,128 @@ getRasterLayerNames <- function(startDateTime, endDateTime, by, variables,
   return(layerNames)
 }
 
-#' Get all active FMI weather stations.
+# function fmi_station()
+
+# Reading the local version included within the package is separated into its
+# own function, since it's also used for tests.
+.fmi_stations_local <- function() {
+  system.file("extdata", "fmi_stations.csv", package="fmi") %>%
+    utils::read.table(header=TRUE, sep=";", as.is=TRUE) %>%
+    tibble::as_tibble()
+}
+# Use a closure for function fmi_station() in order to cache the results.
+.fmi_stations_closure <- function() {
+  cached_stations <- NULL
+  function (groups=NULL, quiet=FALSE) {
+    stations <- NULL
+    if (!is.null(cached_stations)) {
+      stations <- cached_stations
+    } else {
+      tryCatch({
+        installed_packages <- rownames(utils::installed.packages())
+        if (any(c("rvest", "XML") %in% installed_packages)) {
```

Comment:
Single implementation with either `XML` or `rvest/xml2` would've been enough, but since these are in `Suggests` I see no harm in having both.